### PR TITLE
Addon edgexfoundry to edge cluster

### PR DESCRIPTION
**Pull request by OmigaXm**
What type of PR is this?
Adding function

What this PR does:
Add the function to install the components of edgexfoundry to the edge cluster

Which issue(s) this PR fixes:

Fixes #

Special notes for your reviewer:
Edgexfoundry is a useful tool to deal with the issue of manging the things on the edge of the cloud. Installing the components of edgexfoundry easily can help users to use the edgexfoundry to do the task without worring about how to download and install the components on the edge clusters. So we preset the yaml file of the components of edgexfoundry in the project and create or update the yaml file to install when the command 'edgeadm addon edgex' is called. And they can be uninstall by 'edgeadm detach edgex'. Each command supports to install or uninstall specific components of edgex with flags. And it is proved to be able to run correctly.

### CODE REVIEW

**Review by attlee-wang — COMMENTED** [review]
/lgtm

File: pkg/edgeadm/common/common.go [hunk 1]
```diff
@@ -20,8 +20,10 @@ import (
 	"context"
 	"encoding/base64"
 	"fmt"
+	"github.com/superedge/superedge/pkg/edgeadm/constant/manifests/edgex"
 	"k8s.io/klog/v2"
```
Comment (lianghao208) [inline pkg/edgeadm/common/common.go:24]: Can you put these two lines in the k8s.io and github.com/superedge/superedge import group?

File: pkg/edgeadm/constant/const.go [hunk 2]
```diff
@@ -123,3 +124,16 @@ const ApplicationGridWrapperServiceAddr = "http://127.0.0.1:51006"
...
+	Configmap   = 0
+	App         = 1
+	Core        = 2
+	Device      = 3
+	Support     = 4
+	Sysmgmt     = 5
+	Ui          = 6
+	Mqtt        = 7
+	Completely  = 8
+)
```
Comment (lianghao208) [inline pkg/edgeadm/constant/const.go:139]: maybe iota?
```
const (
	Configmap = iota
	App
	Core
	Device
	Support
	Sysmgmt
	Ui
	Mqtt
	Completely
        SerCount
)
```

**Comment by OmigaXm** [comment]
好的 我来修改



---Original---
From: ***@***.***&gt;
Date: Mon, Aug 30, 2021 18:13 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [superedge/superedge] Addon edgexfoundry to edge cluster (#229)




 
@lianghao208 commented on this pull request.
 
 
In pkg/edgeadm/common/common.go:
 &gt; +"github.com/superedge/superedge/pkg/edgeadm/constant/manifests/edgex"  "k8s.io/klog/v2"  
Can you put these two lines in the k8s.io and github.com/superedge/superedge import group?
 
 
In pkg/edgeadm/constant/const.go:
 &gt; +const ( +SerCount    = 9 +Configmap   = 0 +App         = 1 +Core        = 2 +Device      = 3 +Support     = 4 +Sysmgmt     = 5 +Ui          = 6 +Mqtt        = 7 +Completely  = 8 +)  
maybe iota?
 const ( Configmap = iota App Core Device Support Sysmgmt Ui Mqtt Completely         SerCount )  
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

File: pkg/edgeadm/constant/const.go (hunk 2 above)
Comment (neweyes) [reply pkg/edgeadm/constant/const.go:139]: +1

**Comment by attlee-wang** [comment]
Document Push to another PR

**Review by neweyes — APPROVED** [review]

**Pull request merged**